vendor/riscv/riscv-isa-sim/riscv/processor.cc

```diff
@@ -643,7 +644,7 @@ void processor_t::set_mmu_capability(int cap)
 void processor_t::take_interrupt(reg_t pending_interrupts)
 {
   // Do nothing if no pending interrupts
-  if (!pending_interrupts) {
+  if (!pending_interrupts || !interrupt_allowed) {
```

Comment:
Is the default case for `interrupt_allowed` handled such that other users won't get surprised by this?

Reply:
Added a default value now

Reply:
But is it possible for the surrounding logic to change this signal so that users of other processor cores get their Spike-based simulations tampered with unexpectedly?

Reply:
Since `interrupt_allowed` is set to true by default and only modified by the added `interrupt()` method, this should not influence old behavior.